lib/spack/spack/spec.py

```diff
@@ -2878,6 +2878,13 @@ def concretize(self, tests=False):
         else:
             self._old_concretize(tests)
 
+        # Add git version lookup info to concrete Specs (this is generated for
+        # abstract specs as well but the Versions may be replaced during the
+        # concretization process)
+        for spec in self.traverse():
+            if spec.version.is_commit:
+                spec.version.generate_commit_lookup(spec.fullname)
```

Comment:
Should this be here? I'm thinking it might belong in the `SpecBuilder` interface in `spack.solver.asp`

Reply:
I prefer it here personally: IMO the `asp` module is about directly constructing the spec with the needed properties based on concretization constraints; I think it would be preferable to "finalize" properties that don't directly depend on this process outside of that module.

Admittedly, `SpecBuilder.build_specs` has a mixture of those (e.g. calling `spack.spec.Spec.ensure_no_deprecated(s)` isn't strictly required to produce the `Spec`), so this wouldn't be the start of that trend. I can move this there if you prefer.

Reply:
On further thought, I think it does need to move into the SpecBuilder, because not all concretization code paths call `Spec.concretize`. For example, co-concretization of environments does not.